enterprise/service/license.go

```diff
@@ -0,0 +1,152 @@
+package service
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/bytebase/bytebase/api"
+	enterpriseAPI "github.com/bytebase/bytebase/enterprise/api"
+	"github.com/bytebase/bytebase/enterprise/config"
+	"github.com/golang-jwt/jwt/v4"
+	"go.uber.org/zap"
+)
+
+// licenseService is the service for exterprise license.
+type licenseService struct {
+	l      *zap.Logger
+	config *config.Config
+}
+
+// validPlans is a string array of valid plan types.
+var validPlans = []string{
+	api.TEAM.String(),
+	api.ENTERPRISE.String(),
+}
+
+// NewLicenseService will create a new enterprise license service.
+func NewLicenseService(l *zap.Logger, dataDir string, mode string) (*licenseService, error) {
+	config, err := config.NewConfig(l, dataDir, mode)
+	if err != nil {
+		return nil, err
+	}
+
+	return &licenseService{
+		config: config,
+		l:      l,
+	}, nil
+}
+
+// StoreLicense will store license into file.
+func (s *licenseService) StoreLicense(tokenString string) error {
+	return s.writeLicense(tokenString)
+}
+
+// ParseLicense will valid and parse license from file.
+func (s *licenseService) ParseLicense() (*enterpriseAPI.License, error) {
+	tokenString, err := s.readLicense()
+	if err != nil {
+		return nil, err
+	}
+
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+
+		kid, ok := token.Header["kid"].(string)
+		if !ok || kid != s.config.Version {
+			return nil, fmt.Errorf("version '%v' is not valid. expect %s", token.Header["kid"], s.config.Version)
+		}
+
+		key, err := jwt.ParseRSAPublicKeyFromPEM([]byte(s.config.PublicKey))
+		if err != nil {
+			return nil, err
+		}
+
+		return key, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok || !token.Valid {
+		return nil, fmt.Errorf("invalid token")
+	}
+
+	return s.parseClaims(claims)
+}
+
+// parseClaims will valid and parse JWT claims to license instance.
+func (s *licenseService) parseClaims(claims jwt.MapClaims) (*enterpriseAPI.License, error) {
+	err := claims.Valid()
+	if err != nil {
+		return nil, err
+	}
+
+	exp, ok := claims["exp"].(int64)
+	if !ok {
+		return nil, fmt.Errorf("exp is not valid, found '%v'", claims["exp"])
+	}
+	if exp <= time.Now().Unix()/1000 {
```

Comment:
Unix() already returned seconds, why / 1000, similar in Line 122.

If the license generating logic does not use second, we should change that to the second unit.

Reply:
My fault, it's a bug here, the exp should be seconds just like `Unix()`